enos/enos-globals.hcl

```diff
@@ -16,20 +16,43 @@ globals {
     "ent.hsm.fips1402" = ["ui", "enterprise", "cgo", "hsm", "fips", "fips_140_2", "ent.hsm.fips1402"]
   }
   config_modes    = ["env", "file"]
+  consul_editions = ["ce", "ent"]
   consul_versions = ["1.14.11", "1.15.7", "1.16.3", "1.17.0"]
-  distros         = ["ubuntu", "rhel"]
+  distros         = ["amzn2", "leap", "rhel", "sles", "ubuntu"]
+  # Different distros may require different packages, or use different aliases for the same package
+  distro_packages = {
+    amzn2 = ["nc"]
+    leap  = ["netcat", "openssl"]
+    rhel  = ["nc"]
+    # When installing Vault RPM packages on a SLES AMI, the openssl package provided
+    # isn't named "openssl, which rpm doesn't know how to handle. Therefore we add the
+    # "correctly" named one in our package installation before installing Vault.
+    sles   = ["nc", "openssl"]
+    ubuntu = ["netcat"]
+  }
   distro_version = {
-    "rhel"   = var.rhel_distro_version
-    "ubuntu" = var.ubuntu_distro_version
+    "amzn2"  = var.distro_version_amzn2
+    "leap"   = var.distro_version_leap
+    "rhel"   = var.distro_version_rhel
+    "sles"   = var.distro_version_sles
+    "ubuntu" = var.distro_version_ubuntu
   }
   editions = ["ce", "ent", "ent.fips1402", "ent.hsm", "ent.hsm.fips1402"]
-  packages = ["jq"]
-  distro_packages = {
-    ubuntu = ["netcat"]
-    rhel   = ["nc"]
+  package_manager = {
+    "amzn2"  = "yum"
+    "leap"   = "zypper"
+    "rhel"   = "yum"
+    "sles"   = "zypper"
+    "ubuntu" = "apt"
   }
+  packages = ["jq"]
   sample_attributes = {
-    aws_region = ["us-east-1", "us-west-2"]
+    aws_region            = ["us-east-1", "us-west-2"]
+    distro_version_amzn2  = ["2"]
+    distro_version_leap   = ["15.4", "15.5"]
```

Comment:
Great use of sample attrs. I'm still trying to figure out when it's better to use samples attrs or a matrix variant, but it feels like to me _if it doesn't change logic_ it should be a sample attr going forward.